### PR TITLE
Add new function related BBEs

### PR DESCRIPTION
**Pull request by chiranSachintha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.bal [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import ballerina/io;
+
+function subString(string str, int 'start = 0, int end = str.length()) returns string {
+    return str.substring('start, end);
+
```
Comment (rdulmina) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.bal:5]:
```suggestion
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.bal [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+public function main() {
+    // Call the `subString` function with using defaults values of parameters with names `start` and `end`.
```
Comment (KRVPerera) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.bal:9]:
```suggestion
    // Call the `subString` function using default values of `start` and `end` parameters.
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.bal [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    io:println(subString("Ballerina Language"));
+
+    // Call the `subString` function by passing value for default parameter with name `start`.
```
Comment (KRVPerera) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.bal:12]:
```suggestion
    // Call the `subString` function by passing a value for the default parameter `start`.
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.bal [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    io:println(subString("Ballerina Language", 10));
+
+    // Call the `subString` function by passing values for parameters with name `start` and `end`.
```
Comment (KRVPerera) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.bal:15]:
```suggestion
    // Call the `subString` function by passing values for `start` and `end` default parameters.
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Default values for function parameters
+
+Ballerina allows specifying default values for function parameters. You can use any expression such as a literal, or a function call as the default value of a parameter. Additionally, the default value expressions can use the values of preceding parameters.
```
Comment (KRVPerera) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.md:3]:
```suggestion
Ballerina allows specifying default values for function parameters. You can use any expression such as a literal or a function call as the default value of a parameter. Additionally, the default value expressions can use the values of preceding parameters.
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.metatags [hunk 6]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates default values for function parameters in Ballerina.
```
Comment (KRVPerera) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.metatags:1]:
```suggestion
description: This BBE demonstrates how to provide and use the default value of function parameters in Ballerina.
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.metatags [hunk 7]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina, ballerina by example, bbe, default value
```
Comment (KRVPerera) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.metatags:2]:
```suggestion
keywords: ballerina, ballerina by example, bbe, default value, functions, 
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.metatags [hunk 8]
```diff
@@ -0,0 +1,2 @@
+description: This BBE demonstrates call functions with named arguments in Ballerina.
+keywords: ballerina, ballerina by example, bbe, named arguments
```
Comment (KRVPerera) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.metatags:2]:
```suggestion
keywords: ballerina, ballerina by example, bbe, named arguments, functions
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+public function main() {
+    // call `foo` function  using the positional arguments.
+    foo(1, 2, 3);
+
+    // call `foo` function using the named arguments in order of the parameters of the function definition.
+    foo(x = 1, y = 2, z = 3);
+
+    // call `foo` function using the named arguments in order to being different from the parameters of the function definition.
```
Comment (KRVPerera) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal:14]:
```suggestion
    // call `foo` function using the named arguments in a different order from the order of parameters of the function definition.
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+function foo(int x, int y, int z) {
+    io:println("Sum of x, y and z:", x + y + z);
+}
+
+public function main() {
+    // call `foo` function  using the positional arguments.
+    foo(1, 2, 3);
+
+    // call `foo` function using the named arguments in order of the parameters of the function definition.
```
Comment (KRVPerera) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal:11]:
```suggestion
    // call `foo` function using the named arguments in the same order as the parameters of the function definition.
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    foo(x = 1, y = 2, z = 3);
+
+    // call `foo` function using the named arguments in order to being different from the parameters of the function definition.
+    foo(z = 3, y = 2, x = 1);
+
+    // call `foo` function  using the combine of named arguments and positional arguments,
```
Comment (KRVPerera) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal:17]:
```suggestion
    // call `foo` function  using a combination of named arguments and positional arguments.
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# Provide function arguments by name
+
+Ballerina allows you to call functions with named arguments, and these named arguments do not have to be in the same order as the parameters.
```
Comment (KRVPerera) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.md:3]:
```suggestion
Ballerina allows you to call functions with named arguments, which do not have to be in the same order as the parameters.
```

Comment (KRVPerera) [reply examples/provide-function-arguments-by-name/provide_function_arguments_by_name.md:3]: May be we can skip this.

**Review by KRVPerera — COMMENTED** [review]
LGTM

**Comment by MaryamZi** [comment]
@chiranSachintha, can you please resolve the conflict?

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.bal [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import ballerina/io;
+
+function subString(string str, int 'start = 0, int end = str.length())
+                                returns string {
+    return str.substring('start, end);
+}
+
+public function main() {
+    // Call the `subString` function using default values of `start` and `end` parameters.
```
Comment (praneesha) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.bal:9]:
```suggestion
    // Call the `subString` function using the default values of `start` and `end` parameters.
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.bal [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+    io:println(subString("Ballerina Language"));
+
+    // Call the `subString` function by passing a value for the default parameter `start`.
```
Comment (praneesha) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.bal:12]:
```suggestion
    // Call the `subString` function by passing a value for the `start` default parameter.
```

File: examples/default-values-for-function-parameters/default_values_for_function_parameters.md [hunk 15]
```diff
@@ -0,0 +1,7 @@
+# Default values for function parameters
+
+Ballerina allows specifying default values for function parameters. You can use any expression such as a literal or a function call as the default value of a parameter. Additionally, the default value expressions can use the values of preceding parameters.
```
Comment (praneesha) [inline examples/default-values-for-function-parameters/default_values_for_function_parameters.md:3]:
```suggestion
Ballerina allows specifying default values for function parameters. You can use any expression such as a literal or a function call as the default value of a parameter. Additionally, the default value expressions can use the values of the preceding parameters.
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal [hunk 16]
```diff
@@ -0,0 +1,19 @@
+import ballerina/io;
+
+function foo(int x, int y, int z) {
+    io:println("Sum of x, y and z:", x + y + z);
+}
+
+public function main() {
+    // call `foo` function  using the positional arguments.
```
Comment (praneesha) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal:8]:
```suggestion
    // Call the `foo` function using the positional arguments.
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+    foo(1, 2, 3);
+
+    // call `foo` function using the named arguments in the same order as the parameters of the function definition.
```
Comment (praneesha) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal:11]:
```suggestion
    // Call the `foo` function using the named arguments in the same order as the parameters of the function definition.
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal [hunk 18]
```diff
@@ -0,0 +1,19 @@
...
+    foo(x = 1, y = 2, z = 3);
+
+    // call `foo` function using the named arguments in a different order from the order of the parameters in the function definition.
```
Comment (praneesha) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal:14]:
```suggestion
    // Call the `foo` function using the named arguments in a different order from the order of the parameters in the function definition.
```

File: examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+    foo(z = 3, y = 2, x = 1);
+
+    // call `foo` function  using a combination of named arguments and positional arguments.
```
Comment (praneesha) [inline examples/provide-function-arguments-by-name/provide_function_arguments_by_name.bal:17]:
```suggestion
    // Call the `foo` function using a combination of named arguments and positional arguments.
```

**Review by praneesha — APPROVED** [review]

File: examples/index.json [hunk 20]
```diff
@@ -252,13 +266,6 @@
                 "verifyOutput": true,
                 "isLearnByExample": true
             },
-            {
```
Comment (MaryamZi) [inline examples/index.json:255]: There seem to be invalid changes.

**Review by MaryamZi — COMMENTED** [review]
@chiranSachintha, please resolve conflicts.

**Review by gimantha — APPROVED** [review]
LGTM

**Review by rdulmina — APPROVED** [review]

**Pull request merged**